runtime/tests/tests.rs

```diff
@@ -222,35 +280,441 @@ fn generic_asset_transfer_works_with_fee_exchange() {
 
 #[test]
 fn contract_fails() {
-	// Contract itself fails
+	ExtBuilder::default()
+		.initial_balance(1_000_000 * TransactionBaseFee::get())
+		.gas_price(1)
+		.build()
+		.execute_with(|| {
+			let (contract_address, _) = setup_contract(mock::contracts::CONTRACT_WITH_TRAP, dave());
+
+			// Call the newly instantiated contract. The contract is expected to dispatch a call
+			// and then trap.
+			let contract_call = Call::Contracts(pallet_contracts::Call::call(
+				contract_address, // newly created contract address
+				0,                // transfer value in
+				1_000_000,        // gas limit
+				vec![],
+			));
+			let contract_call_extrinsic = sign(CheckedExtrinsic {
+				signed: Some((bob(), signed_extra(2, 0, None, None))),
+				function: contract_call,
+			});
+
+			assert!(Executive::apply_extrinsic(contract_call_extrinsic).is_err());
+		});
+}
+
+// Scenario:
+// - Extrinsic made with a contract call and fee payment in CENNZ
+// - Contract will dispatch a runtime call to move the callers CENNZ funds which should be used for payment
+// This must fail!
+#[test]
+fn contract_dispatches_runtime_call_funds_are_safu() {
+	ExtBuilder::default()
+		.initial_balance(100_000_000_000_000_000_000_000_000) // 1_000_000_000_000 * DOLLARS
```

Comment:
This feels so wrong but don't have a better idea on how to test with sane numbers right now

Reply:
Let me try scale the numbers down a bit

Reply:
This is somewhat addressed in the latest commit